internal/resource/config/oauth/oauth_auth_server_settings_data_source.go

```diff
@@ -0,0 +1,620 @@
+	state.RefreshRollingInterval = types.Int64Value(r.RefreshRollingInterval)
+	state.PersistentGrantReuseGrantTypes = internaltypes.GetStringSet(r.PersistentGrantReuseGrantTypes)
+	state.BypassAuthorizationForApprovedGrants = types.BoolPointerValue(r.BypassAuthorizationForApprovedGrants)
+	state.AllowUnidentifiedClientROCreds = types.BoolPointerValue(r.AllowUnidentifiedClientROCreds)
+	state.AllowUnidentifiedClientExtensionGrants = types.BoolPointerValue(r.AllowUnidentifiedClientExtensionGrants)
+	state.AdminWebServicePcvRef, respDiags = resourcelink.ToDataSourceState(ctx, r.AdminWebServicePcvRef)
+	diags.Append(respDiags...)
+	state.AtmIdForOAuthGrantManagement = types.StringPointerValue(r.AtmIdForOAuthGrantManagement)
+	state.ScopeForOAuthGrantManagement = types.StringPointerValue(r.ScopeForOAuthGrantManagement)
+	state.AllowedOrigins = internaltypes.GetStringSet(r.AllowedOrigins)
+	state.UserAuthorizationUrl = types.StringPointerValue(r.UserAuthorizationUrl)
+	state.RegisteredAuthorizationPath = types.StringValue(r.RegisteredAuthorizationPath)
+	state.PendingAuthorizationTimeout = types.Int64Value(r.PendingAuthorizationTimeout)
+	state.DevicePollingInterval = types.Int64Value(r.DevicePollingInterval)
+	state.ActivationCodeCheckMode = types.StringPointerValue(r.ActivationCodeCheckMode)
+	state.BypassActivationCodeConfirmation = types.BoolValue(r.BypassActivationCodeConfirmation)
+	state.UserAuthorizationConsentPageSetting = types.StringPointerValue(r.UserAuthorizationConsentPageSetting)
+	state.UserAuthorizationConsentAdapter = types.StringPointerValue(r.UserAuthorizationConsentAdapter)
+	state.ApprovedScopesAttribute = types.StringPointerValue(r.ApprovedScopesAttribute)
+	state.ApprovedAuthorizationDetailAttribute = types.StringPointerValue(r.ApprovedAuthorizationDetailAttribute)
+	state.ParReferenceTimeout = types.Int64PointerValue(r.ParReferenceTimeout)
+	state.ParReferenceLength = types.Int64PointerValue(r.ParReferenceLength)
+	state.ParStatus = types.StringPointerValue(r.ParStatus)
+	state.ClientSecretRetentionPeriod = types.Int64PointerValue(r.ClientSecretRetentionPeriod)
+	state.JwtSecuredAuthorizationResponseModeLifetime = types.Int64PointerValue(r.JwtSecuredAuthorizationResponseModeLifetime)
+	return diags
+}
+
+// Read resource information
+func (r *oauthAuthServerSettingsDataSource) Read(ctx context.Context, req datasource.ReadRequest, resp *datasource.ReadResponse) {
+	var state oauthAuthServerSettingsDataSourceModel
+
+	diags := req.Config.Get(ctx, &state)
+	resp.Diagnostics.Append(diags...)
+	if resp.Diagnostics.HasError() {
+		return
+	}
+
+	apiReadOauthAuthServerSettings, httpResp, err := r.apiClient.OauthAuthServerSettingsAPI.GetAuthorizationServerSettings(config.ProviderBasicAuthContext(ctx, r.providerConfig)).Execute()
+
+	if err != nil {
+		config.ReportHttpError(ctx, &resp.Diagnostics, "An error occurred while getting the OAuth Auth Server Settings", err, httpResp)
+		return
+	}
+
+	// Log response JSON
```

Comment:
I believe we decided to remove these marshal json calls, because they don't really do anything at the moment

Reply:
Do you mean for just datasources or resources as well? Because they are in all of the current resources and datasources.

Reply:
Ah I see we made an issue for it (1309) but haven't gone through and removed them yet. Feel free to remove this or just leave it in, since we can handle it as a part of that issue.